src/OmniSharp.Host/CommandLineApplication.cs

```diff
@@ -62,15 +65,6 @@ public int Execute(string[] args)
             return Application.Execute(args.Except(OtherArgs).ToArray());
         }
 
-        public void OnExecute(Func<Task<int>> func)
```

Comment:
Was it necessary to remove this method?

Reply:
The method was using a non-existing API from the new version.
And since the OnExecute is seemingly an unused method (public though), I removed it instead.
I suppose the "Host" is not a library, that was my assumption.

Reply:
@filipw or @david-driscoll may know whether we need to shim in a replacement.

Reply:
it is published on NuGet. I would prefer to stick to not ever having breaking changes and add a shim - thanks

Reply:
I have re-added the method, but made it obsolete. Added a new variant that uses the cancellation token that the new API requires.